### PR TITLE
OCPBUGSM-29306 Allow creating SNO cluster with version 4.8.0-fc.X

### PR DESCRIPTION
**Pull request by eranco74**

### CODE REVIEW

**Review by nmagnezi — CHANGES_REQUESTED** [review]
Worth adding unit tests for `verifyMinimalOpenShiftVersionForSingleNode`

**Comment by danielerez** [comment]
Looks good. Probably worth validating in the subsystem tests as well (kubeapi_test).

File: internal/bminventory/inventory_test.go [hunk 1]
```diff
@@ -2083,6 +2083,61 @@ var _ = Describe("cluster", func() {
...
+				NewClusterParams: &models.ClusterCreateParams{
+					Name:                 swag.String("some-cluster-name"),
+					OpenshiftVersion:     swag.String(insufficientOpenShiftVersionForNoneHA),
+					PullSecret:           swag.String("{\"auths\":{\"cloud.openshift.com\":{\"auth\":\"dG9rZW46dGVzdAo=\",\"email\":\"coyote@acme.com\"}}}"),
+					HighAvailabilityMode: &noneHaMode,
+				},
+			})
+			verifyApiError(reply, http.StatusBadRequest)
+		})
+		It("create non ha cluster success, release version is grater than minimal", func() {
```
Comment (danielerez) [inline internal/bminventory/inventory_test.go:2101]: s/grater/greater

File: internal/bminventory/inventory_test.go [hunk 2]
```diff
@@ -2083,6 +2083,61 @@ var _ = Describe("cluster", func() {
...
+				},
+			})
+			Expect(reflect.TypeOf(reply)).Should(Equal(reflect.TypeOf(installer.NewRegisterClusterCreated())))
+			actual := reply.(*installer.RegisterClusterCreated)
+			Expect(actual.Payload.HighAvailabilityMode).To(Equal(swag.String(noneHaMode)))
+			Expect(actual.Payload.UserManagedNetworking).To(Equal(swag.Bool(true)))
+			// verify VipDhcpAllocation was set to false even though it was sent as true
+			Expect(actual.Payload.VipDhcpAllocation).To(Equal(swag.Bool(false)))
+		})
+		It("create non ha cluster success, release version is pre-release and grater than minimal", func() {
```
Comment (danielerez) [inline internal/bminventory/inventory_test.go:2121]: s/grater/greater

File: internal/bminventory/inventory_test.go [hunk 3]
```diff
@@ -2083,6 +2083,61 @@ var _ = Describe("cluster", func() {
...
+					HighAvailabilityMode: &noneHaMode,
+				},
+			})
+			Expect(reflect.TypeOf(reply)).Should(Equal(reflect.TypeOf(installer.NewRegisterClusterCreated())))
+			actual := reply.(*installer.RegisterClusterCreated)
+			Expect(actual.Payload.HighAvailabilityMode).To(Equal(swag.String(noneHaMode)))
+			Expect(actual.Payload.UserManagedNetworking).To(Equal(swag.Bool(true)))
+			// verify VipDhcpAllocation was set to false even though it was sent as true
+			Expect(actual.Payload.VipDhcpAllocation).To(Equal(swag.Bool(false)))
+		})
```
Comment (danielerez) [inline internal/bminventory/inventory_test.go:2140]: nit: maybe also a test for release version equals minimal

Comment (eranco74) [reply internal/bminventory/inventory_test.go:2140]: That is the first tc (happy flow)

**Comment by eranco74** [comment]
> Looks good. Probably worth validating the subsystem tests as well (kubeapi_test).
OK, I'll add it in another PR, I'd like this fix to get merged fist.
Adding subsystem might take some time.

**Comment by danielerez** [comment]
> > Looks good. Probably worth validating the subsystem tests as well (kubeapi_test).
> > OK, I'll add it in another PR, I'd like this fix to get merged first.
> > Adding subsystem might take some time.

Sure. I can handle the subsystem as part of testing the scenario of an existing version with full release version format.

**Pull request merged**